### ISSUE TITLE
Deprecated depends

### ISSUE THREAD

**Issue by Fantu** [linuxmint/cinnamon#11173]
Hi I'm looking about some deps deprecated, I found there is gir1.2-gtkclutter-1.0 that also take have old not maintained and deprecated clutter and cogl (out of muffin) as deps.
From a fast search I don't found import of gtkclutter so seems not used anymore, is it correct? and clutter cinnamon should now use only the muffin one, is it correct?
There is also caribou that is not maintained upstream anymore (only security and fail to build fix in latest years), latest upstream version of 6 years ago and still use the deprecated clutter (out of muffin/mutter).
I think can be good search for alternative virtual keyboard. @mtwebster @clefebvre what do you think?

**Comment by mtwebster** [comment]
Yeah we can get rid of gtkclutter from cinnamon.  Unfortunately it's still needed in nemo-preview, but that's another issue.

We've talked about replacing or absorbing libcaribou - I've never found a similar library though. Gnome went and implemented their own virtual keyboard in gnome-shell, which I really don't  want us to have to do (I think they did because caribou wouldn't be useful in wayland without a lot of work).

We have onboard also by default, but it really doesn't integrate well at all, but is more useful for applications.

**Comment by lestcape** [comment]
> I think they did because caribou wouldn't be useful in Wayland without a lot of work

Can be implemented with Gtk3, but the problem will be with Gtk4. It need the same type of solution that what will need mate-panel to work with Gtk4 in Wayland. See: https://gitlab.gnome.org/GNOME/gtk/-/issues/2132

So, if you want to have Wayland and port all to Gtk4 some day, you should considered avoid the problem that Mate already face. Is cleared that GNOME have not any interest in resolved that any time soon. A PopOS developer have more than 3 years trying to push that in Gtk...

If he have success, the in screen keyboard can be implemented at the top of gtk-layer-shell. More info:
https://github.com/wmww/gtk-layer-shell/issues/37

**Comment by Fantu** [comment]
there is a discussion to remove caribou from debian, remained only cinnamon using it: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1041641
@clefebvre @mtwebster can you please tell what do you think relating of the future of virtual keyboard in cinnamon?

**Comment by diizzyy** [comment]
As well for FreeBSD, https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=276955

Friendly ping on this issue

**Comment by City-busz** [comment]
Cinnamon developers should fork Caribou if they want to continue using it.

**Comment by Fantu** [comment]
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1093210

Is there any news about on-screen keyboard possible replacement?

@leigh123linux Fedora what think about caribou? Not had fails to build with gcc-14? On Debian bugtracker I saw a patch for possible fix today but I don't know when I'll have to try
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1074863

**Comment by leigh123linux** [comment]
@Fantu  It can be fixed by running this


```
find -name '*.vala' -exec touch {} \;
```

https://bugzilla.redhat.com/show_bug.cgi?id=2259130#c1

**Comment by leigh123linux** [comment]
It still builds ok with gcc-15

https://koji.fedoraproject.org/koji/buildinfo?buildID=2617795

**Comment by Fantu** [comment]
@leigh123linux thanks for reply
Today @smcv fixed build successfull removing generated vala files: https://salsa.debian.org/gnome-team/caribou/-/commits/debian/latest/?ref_type=HEADS
I'll try it this weekend if I'll have time

**Comment by Fantu** [comment]
A good news: seems there is a work in progress for the [new onscreen keyboard](https://github.com/linuxmint/cinnamon/commit/1d00531ca7c74d6948a176a8c2fd671ba74bf4ba) that will replace caribou. Thanks to @mtwebster for the big work.
